pkg/controllers/status/cluster_status_controller.go

```diff
@@ -82,6 +78,10 @@ type ClusterStatusController struct {
 	ClusterLeaseRenewIntervalFraction float64
 	// ClusterLeaseControllers store clusters and their corresponding lease controllers.
 	ClusterLeaseControllers sync.Map
+	// ClusterFailureThreshold is the duration of failure for the cluster to be considered unhealthy.
+	ClusterFailureThreshold metav1.Duration
+	// ClusterFailureTimestamp is the timestamp of the first failed cluster health check.
+	ClusterFailureTimestamp map[string]*time.Time
```

Comment:
I guess we'll add a success threshold too, then we need to introduce another map? as well as a lock?

Reply:
Not really, we can change the map value from `time` to `{successTime, failureTime}`